ramls/reserve.json

```diff
@@ -154,6 +154,10 @@
                     "type": "string",
                     "description": "The HRID of the associated instance record"
                 },
+                "instanceDiscoverySuppress": {
```

Comment:
Does it seem weird to you that this property of the instance is copied into a reserve of an item that is in a holding of that instance? Oh well, I guess someone had a good reason for wanting it this way.

Reply:
It seems **very** weird to me but looking at the rest of the contents it fits the existing model better... Like, say we were even going to decide "let's break out the instance/holding stuff to `copiedInstance` and `copiedHolding` because it's more semantically accurate!"

Well then you'd have the issue of what the hell to do with the location info, because the temporary location in that object can either be from the item or the holding, and same with the physical one... So then we'd need frontend updates if we want everything to be _truly_ semantically accurate. 🤦 

Honestly I can't friggin wait until we have Kafka or something to avoid all the variously shaped and sized hoops we jump through to deal with data from foreign microservices.

Reply:
I'm not seeing how Kafka is going to help with this _at all_.

Reply:
My thought was that it might've helped had it always been around and the module built around it. That's because - at least for me - it lessens the amount of "oh it's just one property" shortcuts I tend to take when storing foreign data. 

E.g., I'm subscribing to `instances`, `holdings` and `items` topics and I need to store data from them all in my own local data storage (for searching or passing-on or whatever). I'm probably gonna start by looking at a solution where I have my own "tables" (or whatever analog) for those topics just by virtue of the fact that I'm gonna have to wire in all the update events and everything, rather than smooshing them all into one blob so that when I handle a `holding_edited` event, I'm not touching the store that holds the items and instances.

Maybe it's just me :P

Reply:
Well, the purpose of `copiedItem` is to make it possible to search for a reserve by field of the reserved item. To do that you have to store that data in the reserve one way or another. The current mechanism is just copying the item when the reserve is placed. Are you merely proposing that Kafka would have given us a more elegant method of copying the relevant data? Or proposing a more radically different solution that is as yet opaque to me?

(I note in passing that Kafka causes _endless_ problems in production. I don't know why it should be so, but it is.)